weavelet.go

```diff
@@ -110,6 +97,9 @@ func newWeavelet(ctx context.Context, componentInfos []*codegen.Registration) (*
 		tcpClients:       map[string]*client{},
 		loads:            map[string]*loadCollector{},
 	}
+
+	// TODO(mwhittaker): getEnv starts the WeaveletConn handler which calls
```

Comment:
@spetrovic77 I realized that the `WeaveletConn` has the same annoying set up issues as the `EnvelopeConn`. Here, I want to set up the connection to the envelope, but I don't want to start receiving requests until `w` is fully set up.

Reply:
Would the solution (at least in the short-term) be to move the `WeaveletConn.Serve` call from `newRemoteEnv`, to `weavelet.start` ? That seems to be where other goroutines are started for the weavelet, and at that point `w` would be fully constructed.

That would require adding a new method to the `env` interface that is only implemented by `remoteEnv`. There's already one method that is only used in remote mode (`WeaveletListener`).

If that sounds right, maybe the `WeaveletListener` method on `env` could be replaced by:

```go
type env interface {
    ...
    // remote returns nil in single process mode, and returns the
    // remoteEnv in multi process mode.
   remote() *remoteEnv
```

Or maybe avoid the `remote()` method and instead use a type assertion?

`weavelet.start` could do something like this, while avoiding no-op implementations of these methods on `singleProcessEnv`:

```go
if !w.info.SingleProcess {
    remote := env.(*remoteEnv)
    lis := remote.conn.Listener()
    ...
    go remote.conn.Serve()
```

If this approach sounds interesting, I can prepare a PR based on this one with those changes.

Reply:
This sounds great to me! Definitely feel free to open a PR. 

One thing that's bad about our `Conn` APIs right now is that you cannot issue RPCs until you call `Serve`. Er well, you can, but they block because `Serve` needs to run to process the RPC replies. This means that if you delay calling `Serve`, you have to be very careful not to call any RPCs or else things deadlock. Srdjan is working on improving the API to avoid this.

Reply:
Looks like this is really flakey. @dnephin, we're going to have to fix it ASAP, hope you're not halfway to fixing! :)